src/main.py

```diff
@@ -0,0 +1,117 @@
+from alerts_aggregator import AlertsAggregator
+from oref_alert_indexer import OrefAlertsIndexer  
+from fastapi import FastAPI, HTTPException , Depends, Header
+import pandas as pd
+import requests
+from errors import OrefAPIException, WrongSettlementException, NoAlarmsException, InvalidSettlement
+from models import AlertsQueryInput
+from datetime import datetime, timedelta
+app = FastAPI()
+
+CSV_FILE_PATH = "C:/Projects/Python/Projects/BestTimeToShower/data/data.csv"
+dataframe = pd.read_csv(CSV_FILE_PATH)
+alert_indexer = OrefAlertsIndexer()
+alert_aggregator = AlertsAggregator(dataframe)
+
+def get_api_key(api_key: str = Header(..., convert_underscores=False)):
+
+    if api_key == "red_color":
+        return api_key
+    else:
+        raise HTTPException(status_code=401, detail="Invalid API key")
+
+
+@app.post("/sync_oref_alerts")
+async def get_oref_alert(from_date: str, to_date: str, api_key: str = Depends(get_api_key)):
+    try:
+        global alert_aggregator
+        alert_indexer.arrange_alarms_within_csv(from_date, to_date)
+        df = pd.read_csv(CSV_FILE_PATH)
+        alert_aggregator = AlertsAggregator(df)
```

Comment:
do you remember why you create a new `alert_aggregator` instance here?
document that in the code.

Also, I dont like that idea.
what I want you to do is to create a `reload_data` function under `AlertsAggregator` that will replace the new dataset implace